retrieval_service/datastore/providers/firestore.py

```diff
@@ -35,47 +38,111 @@ class Config(BaseModel, datastore.AbstractConfig):
 
 class Client(datastore.Client[Config]):
     __client: AsyncClient
+    policies_collection: AsyncQuery
+    amenities_collection: AsyncQuery
 
     @datastore.classproperty
     def kind(cls):
         return FIRESTORE_IDENTIFIER
 
     def __init__(self, client: AsyncClient):
         self.__client = client
+        self.policies_collection = AsyncQuery(self.__client.collection("policies"))
+        self.amenities_collection = AsyncQuery(self.__client.collection("amenities"))
 
     @classmethod
     async def create(cls, config: Config) -> "Client":
         return cls(AsyncClient(project=config.projectId))
 
+    async def delete_collections(collection_list: list[AsyncCollectionReference]):
```

Comment:
consider reducing visibility 

reducing visibility means fewer things use this, which makes it easier to change later if needed
```suggestion
    async def __delete_collections(collection_list: list[AsyncCollectionReference]):
```